src/unix/system.c

```diff
@@ -341,7 +344,16 @@ void Sys_Init(void)
     if (!homedir) {
 	    Sys_Error("Homedir not found!\n");
     }
-    sprintf(homegamedir, "%s/%s", homedir, ".quake2rtx");
+    xdg_data_home_dir = getenv("XDG_DATA_HOME");
+    if (xdg_data_home_dir == NULL) {
+	    xdg_data_home_dir = malloc(sizeof(char) * 200);
```

Comment:
You know the length you need to assemble a homedir, no need to arbitrarily choose an allocation size.